src/Serializer/AbstractItemNormalizer.php

```diff
@@ -122,14 +134,33 @@ protected function extractAttributes($object, $format = null, array $context = [
     /**
      * {@inheritdoc}
      */
-    protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
-    {
+    protected function getAllowedAttributes(
+        $classOrObject,
+        array $context,
+        $attributesAsString = false
+    ) {
         $options = $this->getFactoryOptions($context);
-        $propertyNames = $this->propertyNameCollectionFactory->create($context['resource_class'], $options);
+        $propertyNames = $this
+            ->propertyNameCollectionFactory
+            ->create($context['resource_class'], $options);
 
         $allowedAttributes = [];
         foreach ($propertyNames as $propertyName) {
-            $propertyMetadata = $this->propertyMetadataFactory->create($context['resource_class'], $propertyName, $options);
+            $propertyMetadata = $this
+                ->propertyMetadataFactory
+                ->create($context['resource_class'], $propertyName, $options);
+
+            if (
+                isset($context['api_denormalize'])
+                    && !$propertyMetadata->isWritable()
+                    && !$propertyMetadata->isIdentifier()
+            ) {
+                throw new RuntimeException(sprintf(
+                    'Property \'%s.%s\' is not writeable',
+                    $context['resource_class'],
+                    $propertyName
+                ));
+            }
```

Comment:
This needs to be discussed. I'm not sure that when a property is read only, it should throw when we try to write on the resource class. It can be fair to want to keep properties from beeing written no?

We indeed do not have a test for this, and I do agree with you that `Metadata` debugging can be a pain (finding whether a `property` is `readable, readableLink` etc.